contentcuration/contentcuration/tests/test_labels.py

```diff
@@ -0,0 +1,253 @@
+            cats = _random_labels(categories, length_categories)
+            node.gradeLevelStr = ",".join(grade)
+            node.resourceTypeStr = ",".join(res)
+            node.learningActivityStr = ",".join(acts)
+            node.accessibilityStr = ",".join(accs)
+            node.categoryStr = ",".join(cats)
+            node.save()
+        self.elapsed = time() - init_time
+        print(
+            "USING STRING: Assigning random labels to {} nodes took {} seconds".format(
+                len(self.nodes), self.elapsed
+            )
+        )
+
+        init_time = time()
+        strings = len(ContentNode.objects.filter(resourceTypeStr__contains=VIDEO))
+        self.elapsed = time() - init_time
+        print(
+            "USING STRING: Finding for resource label in {} nodes took {} seconds".format(
+                strings, self.elapsed
+            )
+        )
+
+        init_time = time()
+        strings = len(ContentNode.objects.filter(categoryStr__contains=MATH))
+        self.elapsed = time() - init_time
+        print(
+            "USING STRING: Finding for maths and descendants labell in {} nodes took {} seconds".format(
+                strings, self.elapsed
+            )
+        )
+
+    def test_massive_array(self):
+        init_time = time()
+        grades, length_grades = list(GRADE_LEVEL), len(GRADE_LEVEL)
+        resources, length_resources = list(RESOURCE_TYPE), len(RESOURCE_TYPE)
+        activities, length_activities = list(LEARNING_ACTIVITY), len(LEARNING_ACTIVITY)
+        accesibility, length_accessibility = list(ACCESIBILITY), len(ACCESIBILITY)
+        categories, length_categories = list(CATEGORY), len(CATEGORY)
+        for node in self.nodes:
+            grade = _random_labels(grades, length_grades)
+            res = _random_labels(resources, length_resources)
+            acts = _random_labels(activities, length_activities)
+            accs = _random_labels(accesibility, length_accessibility)
+            cats = _random_labels(categories, length_categories)
+
+            node.gradeLevelArray = grade
+            node.resourceTypeArray = res
+            node.learningActivityArray = acts
+            node.accessibilityArray = accs
+            node.categoryArray = cats
+            node.save()
+        self.elapsed = time() - init_time
+        print(
+            "USING ARRAY: Assigning random labels to {} nodes took {} seconds".format(
+                len(self.nodes), self.elapsed
+            )
+        )
+
+        init_time = time()
+        strings = len(ContentNode.objects.filter(resourceTypeArray__contains=[VIDEO]))
+        self.elapsed = time() - init_time
+        print(
+            "USING ARRAY: Finding for resource label in {} nodes took {} seconds".format(
+                strings, self.elapsed
+            )
+        )
+
+        init_time = time()
+        strings = len(ContentNode.objects.filter(categoryArray__icontains=MATH))
```

Comment:
An alternative to this might be to pass in an array of `MATH` and all its descendant categories and instead of `icontains` do `overlap`:
```
strings = len(ContentNode.objects.filter(categoryArray__overlap=[MATH, ....]))
```

Reply:
Done, and also tested it using indexes, and creating the nodes using `TreeBuilder`
Conclusions I can extract with the current logic to search for tags:

- Indexes make things slower
- Using overlap instead of contains is slower in arrays
- When searching for not-categories, i.e. pure arrays without descendants, strings is the fastest format
- When searching for categories, i.e. having descendants, jsonb is the fastest format